src/Files.App/Filesystem/Security/AccessControlEntry.cs

```diff
@@ -0,0 +1,612 @@
+
+		public string AccessMaskFlagsHumanized
+		{
+			get
+			{
+				var accessMaskStrings = new List<string>();
+
+				if (AccessMaskFlags == AccessMaskFlags.NULL)
+				{
+					accessMaskStrings.Add("None".GetLocalizedResource());
+				}
+
+				if (FullControlAccess)
+					accessMaskStrings.Add("SecurityFullControlLabel/Text".GetLocalizedResource());
+				else if (ModifyAccess)
+					accessMaskStrings.Add("SecurityModifyLabel/Text".GetLocalizedResource());
+				else if (ReadAndExecuteAccess)
+					accessMaskStrings.Add("SecurityReadAndExecuteLabel/Text".GetLocalizedResource());
+				else if (ReadAccess)
+					accessMaskStrings.Add("SecurityReadLabel/Text".GetLocalizedResource());
+
+				if (!FullControlAccess && !ModifyAccess && WriteAccess)
+					accessMaskStrings.Add("Write".GetLocalizedResource());
+
+				if (SpecialAccess)
+					accessMaskStrings.Add("SecuritySpecialLabel/Text".GetLocalizedResource());
+
+				return string.Join(",", accessMaskStrings);
+			}
+		}
+
+		public string IsInheritedHumanized
+			=> IsInherited ? "Yes".GetLocalizedResource() : "No".GetLocalizedResource();
+
+		public string InheritanceFlagsHumanized
+		{
+			get
+			{
+				var inheritanceStrings = new List<string>();
+
+				if (PropagationFlags == PropagationFlags.None ||
+					PropagationFlags == PropagationFlags.NoPropagateInherit)
+					inheritanceStrings.Add("SecurityAdvancedFlagsFolderLabel".GetLocalizedResource());
+
+				if (InheritanceFlags.HasFlag(InheritanceFlags.ContainerInherit))
+					inheritanceStrings.Add("SecurityAdvancedFlagsSubfoldersLabel".GetLocalizedResource());
+
+				if (InheritanceFlags.HasFlag(InheritanceFlags.ObjectInherit))
+					inheritanceStrings.Add("SecurityAdvancedFlagsFilesLabel".GetLocalizedResource());
+
+				// Capitalize first letter
+				if (inheritanceStrings.Any())
+					inheritanceStrings[0] = char.ToUpperInvariant(inheritanceStrings[0].First()) + inheritanceStrings[0][1..];
+
+				return string.Join(",", inheritanceStrings);
+			}
+		}
+
+		private bool _isSelected;
+		public bool IsSelected
+		{
+			get => _isSelected;
+			set
+			{
+				if (SetProperty(ref _isSelected, value))
+				{
+					if (!value)
+						AreAdvancedPermissionsShown = false;
+
+					OnPropertyChanged(nameof(IsEditEnabled));
+				}
+			}
+		}
+
+		private bool _areAdvancedPermissionsShown;
+		public bool AreAdvancedPermissionsShown
+		{
+			get => _areAdvancedPermissionsShown;
+			set
+			{
+				// Reconstruct list
+				if (SetProperty(ref _areAdvancedPermissionsShown, value))
+					AccessMaskItemList = GetAllAccessMaskList();
+			}
+		}
+
+		public bool IsEditEnabled
+			=> IsSelected && !IsInherited;
+
+		private List<AccessMaskItem> _accessMaskItemList;
+		public List<AccessMaskItem> AccessMaskItemList
+		{
+			get => _accessMaskItemList;
+			set => SetProperty(ref _accessMaskItemList, value);
+		}
+
+		public RelayCommand<string> ChangeAccessControlTypeCommand { get; set; }
+		public RelayCommand<string> ChangeInheritanceFlagsCommand { get; set; }
```

Comment:
Would those be any interested to be transformed into RichCommands? Might be overkill, but worth asking.